### PR TITLE
Add Input Component

### PR DESCRIPTION
**Pull request by simeonoff**
Closes #44

### CODE REVIEW

File: src/components/input/input.material.scss [hunk 1]
```diff
@@ -0,0 +1,502 @@
...
+            };
+            block: {
+                start-width: $active-border-width;
+                end-width: $active-border-width;
+            };
+        }
+    }
+
+    [part='prefix'] {
+        padding-inline-start: 4px;
```
Comment (SisIvanova) [inline src/components/input/input.material.scss:197]:
```suggestion
        padding-inline-start: rem(4px);
```

File: src/components/input/input.material.scss [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+    [part='prefix'],
+    [part='suffix'],
+    [part='helper-text'] {
+        color: color(gray, 400);
+    }
+
+    [part='start'],
+    [part='filler'],
+    [part='notch'],
+    [part='end'], {
```
Comment (SisIvanova) [inline src/components/input/input.material.scss:375]:
```suggestion
    [part='end'] {
```

**Review by SisIvanova — APPROVED** [review]

**Review by desig9stein — DISMISSED** [review]

File: src/components/input/input.ts [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+import { watch } from '../common/decorators';
+import { EventEmitterMixin } from '../common/mixins/event-emitter.js';
+import { partNameMap } from '../common/util';
+import { SizableMixin } from '../common/mixins/sizable';
+
+let nextId = 0;
+
+type Direction = 'ltr' | 'rtl' | 'auto';
+
+export interface IgcRadioEventMap {
```
Comment (rkaraivanov) [inline src/components/input/input.ts:16]: I guess you meant `InputEventMap`

**Review by rkaraivanov — CHANGES_REQUESTED** [review]
LGTM except for two points:

1. API documentation - Document the public API of the component. Also the `handle${foo}` and `render${part}` methods could be with access level `protected`.
2. What about `focus()` and `blur()` methods? We expose those for the other input types (checkbox, radio). I think they should be present here as well

File: src/components/input/input.ts [hunk 4]
```diff
@@ -41,21 +60,24 @@ export class IgcInputComponent extends SizableMixin(
...
+  /** The direction attribute of the control. */
   @property({ reflect: true })
   dir: Direction = 'auto';
 
+  /** The type attribute of the control. */
   @property({ reflect: true })
   type: 'email' | 'number' | 'password' | 'search' | 'tel' | 'text' | 'url' =
     'text';
 
+  /** The inputmode attribute of the control. */
```
Comment (SisIvanova) [inline src/components/input/input.ts:80]: input mode

File: src/components/input/input.ts [hunk 5]
```diff
@@ -133,27 +171,26 @@ export class IgcInputComponent extends SizableMixin(
...
 
-  resolvePartNames(base: string) {
-    return {
-      [base]: true,
-      prefixed: this._prefixLength > 0,
-      suffixed: this._suffixLength > 0,
-    };
-  }
-
+  /** Checks for validity of the control and shows the browser message if it invalid. */
```
Comment (SisIvanova) [inline src/components/input/input.ts:174]: its

**Review by SisIvanova — APPROVED** [review]

**Review by SisIvanova — APPROVED** [review]

**Review by rkaraivanov — APPROVED** [review]

**Review by DiyanDimitrov — APPROVED** [review]

**Pull request merged**